From f67986a4ee549f588528c493de1dd2d57042b6fc Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 15:21:26 +0800
Subject: [PATCH 260/301] AOSCOS: staging: sb105x: replace alloc_tty_driver()
 with tty_alloc_driver()

Per commit 39b7b42be4a8 ("tty: stop using alloc_tty_driver"), all
instances of `alloc_tty_driver()' were revised into `tty_alloc_driver()'
with an extra `flags' argument (merged from the `flags' member of `struct
tty_driver'. The `tty_alloc_driver()' function was then removed in commit
56ec5880a28e ("tty: drop alloc_tty_driver").

Follow this chain of changes and revise the driver code.

Fixes: "AOSCOS: Revert "staging: sb105x: delete the driver""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/staging/sb105x/sb_pci_mp.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/staging/sb105x/sb_pci_mp.c b/drivers/staging/sb105x/sb_pci_mp.c
index cd98d3b0183d..ccf02cfeb2fc 100644
--- a/drivers/staging/sb105x/sb_pci_mp.c
+++ b/drivers/staging/sb105x/sb_pci_mp.c
@@ -1728,12 +1728,10 @@ static int mp_register_driver(struct uart_driver *drv)
 	}
 	memset(drv->state, 0, sizeof(struct sb_uart_state) * drv->nr);
 
-	normal = alloc_tty_driver(drv->nr);
-	if (!normal)
-	{
-		printk("SB PCI Error: tty allocation error!\n");
-		goto out;
-	}
+	normal = tty_alloc_driver(drv->nr,
+				  TTY_DRIVER_REAL_RAW | TTY_DRIVER_DYNAMIC_DEV);
+	if (IS_ERR(normal))
+		return PTR_ERR(normal);
 
 	drv->tty_driver = normal;
 
@@ -1750,7 +1748,6 @@ static int mp_register_driver(struct uart_driver *drv)
 	normal->subtype		= SERIAL_TYPE_NORMAL;
 	normal->init_termios	= tty_std_termios;
 	normal->init_termios.c_cflag = B9600 | CS8 | CREAD | HUPCL | CLOCAL;
-	normal->flags		= TTY_DRIVER_REAL_RAW | TTY_DRIVER_DYNAMIC_DEV;
 	normal->driver_state    = drv;
 
 	tty_set_operations(normal, &sb_mp_ops);
-- 
2.48.1

